### ISSUE TITLE
Is it possible to use a variable within a path of a secret?

### ISSUE THREAD

**Issue by slurikadre** [twpayne/chezmoi#1624]
## What exactly are you trying to do?

What I'd like to be able to do is:

```
(vault "kv/users/$username/mysecret").data.data.keyName
```

or even:

```
(vault "kv/users/$hostname/mysecret").data.data.keyName
```

## What have you tried so far?

I think this would be a lot neater than:

```
{{- if .personal }}
{{  if eq .chezmoi.hostname "host1" "host2" -}}
{{      (vault "kv/users/user1/mysecret").data.data.keyName }}
{{-   if eq .chezmoi.hostname "host3" "host4"  -}}
{{      (vault "kv/users/user2/mysecret").data.data.keyName }}
{{-   end }}
{{- end }}
```

I don't really care about having a condition for if the hostname isn't found, because vault will throw a fairly obvious error that the secret couldn't be found.

## Where else have you checked for solutions?

* [x] I have read [chezmoi's how-to guide](https://github.com/twpayne/chezmoi/blob/master/docs/HOWTO.md), and not found the answer.
* [x] I have read [chezmoi's FAQ](https://github.com/twpayne/chezmoi/blob/master/docs/FAQ.md), and not found the answer.
* [x] I have searched [chezmoi's reference guide](https://github.com/twpayne/chezmoi/blob/master/docs/REFERENCE.md), and not found the answer.

**Comment by felipecrs** [comment]
You have some options:

```
(vault (printf "kv/users/%s/mysecre" .chezmoi.hostname).data.data.keyName
```

**Comment by slurikadre** [comment]
Thanks this works exactly as intended!

**Comment by slurikadre** [comment]
Thanks, one quick follow up question about choosing the first object. I'm pretty sure it's tied up with that `index` keyword:

I use this JSON data in my Mutt config.

```json
{
  "mail": [
    {
      "realName": "Bill Blogs",
      "username": "bill@billpersonal.com"
      "keyID": "0x12345"
    },
    {
      "realName": "Bill Blogs",
      "username": "bill@billwork.com"
    }
  ]
}
```

I thought seeing as the data is there I'd use it for my git user template

```
{{- if .personal -}}
{{-     template "git/user" dict "uid" .primaryuid }}
{{-   end }}
{{- end }}
```

For that I often want to iterate through the list. However if I only want the first entry bill@billpersonal.com how can I set the `index` to `0`.

I've done it before, but can't seem to figure it out in this situation.

```
{{- range $v, $gitUser := (vault (printf "kv/users/%s/mail" .uid)).data.data.mail }}
	name = {{ .realName }}
	email = {{ .username }}
	signingkey = {{ if eq (hasKey $gitUser "keyID") true }}{{ .keyID }}{{ end }}
{{- end }}
```

In short what I get is: 

```
[user]
	name = Bill Blogs
	email = bill@billpersonal.com
	signingkey = 0x12345
	name = Bill Blogs
	email = bill@billwork.com
	signingkey = 
```

What I want is:

```
[user]
	name = Bill Blogs
	email = bill@billpersonal.com
	signingkey = 0x12345
```

**Comment by slurikadre** [comment]
I think I figured it out:

Get rid of the loop lol.

```
{{ $gitUser := index (vault (printf "kv/users/%s/mail" .uid)).data.data.mail 0}}
	name = {{ $gitUser.realName }}
	email = {{ $gitUser.username }}
	signingkey = {{ if eq (hasKey $gitUser "keyID") true }}{{ $gitUser.keyID }}{{ end }}
```
